src/server/pkg/deploy/assets/assets.go

```diff
@@ -138,6 +138,18 @@ type StorageOpts struct {
 	UploadConcurrencyLimit int
 }
 
+const (
+	// RequireCriticalServersOnlyEnvVar is the environment variable for requiring critical servers only.
+	RequireCriticalServersOnlyEnvVar = "REQUIRE_CRITICAL_SERVERS_ONLY"
```

Comment:
do we want this to be configurable?

Reply:
Yes, to allow users to be unblocked when a noncritical server errors during setup or while running.